framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -894,7 +892,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
+            checkSupportedQuals();
+        }
+        return Collections.unmodifiableSet(supportedQuals);
```

Comment:
You'll also have to remove `final` from the declaration of `supportedQuals`...

The design before your changes was: 1) use `final` to ensure that `supportedQuals` references only 1 copy of a set, and 2) that set is immutable. Together it strongly ensures that the qualifiers supported by an ATF is unchangeable after initializing the ATF.

Now it is weaker; without `final`, the reference _could_ be updated to point to any other set. But since it is a private field this will probably be fine.